src/main/java/com/thebois/models/world/resources/ResourceFactory.java

```diff
@@ -0,0 +1,37 @@
+package com.thebois.models.world.resources;
+
+/**
+ * Factory used to create resources.
+ */
+public final class ResourceFactory {
+
+    private ResourceFactory() {
+
+    }
+
+    /**
+     * Generate a resource with given type at given position.
+     *
+     * @param type What type of resource.
+     * @param x    X position for resource.
+     * @param y    Y position for resource.
+     *
+     * @return The created resource.
+     *
+     * @throws UnsupportedOperationException If the type of resource wanted to be created have not
+     *                                       been implemented in the factory.
+     */
+    public static IResource createResource(final ResourceType type, final int x, final int y) {
+        if (type == ResourceType.WATER) {
+            return new Water(x, y);
+        }
+        else if (type == ResourceType.TREE) {
+            return new Tree(x, y);
+        }
+        else if (type == ResourceType.ROCK) {
+            return new Rock(x, y);
+        }
+        throw new UnsupportedOperationException();
```

Comment:
style

Very minor, but you could've used a switch here instead. Might be more compact, if you use the new switch expression:
```java
  return switch (day) {
        case MONDAY, FRIDAY, SUNDAY -> {
            System.out.println(6);
            yield 6;
        }
        case TUESDAY -> {
            System.out.println(7);
            yield 7;
        }
        case THURSDAY, SATURDAY -> {
            System.out.println(8);
            yield 8;
        }
        case WEDNESDAY -> {
            System.out.println(9);
            yield 9;
        }
        default -> {
            throw new IllegalStateException("Invalid day: " + day);
    }
```

Reply:
Can't really test the default as well since I can't call the method with null as resource type since it breaks the switch.

Reply:
I left the TerrainFactory as it, so we can compare the two, and decide which one we want to use.